### PR TITLE
Allow using "vips" driver

### LINKED ISSUE #1494: Unable to use `vips` driver
Hello, I would like to use `vips` driver (https://packagist.org/packages/rokka/imagine-vips), but config:
```yaml
liip_imagine:
    driver: "vips"
```
fails with:
```
Invalid configuration for path "liip_imagine.driver": Invalid imagine driver specified: "vips"
```

Is there any way to get that working?

### Preconditions

1. Bundle version `2.9.0`;
2. Symfony version `6.1.9`;
1. `vips` implementation for imagine https://packagist.org/packages/rokka/imagine-vips;

### Steps to reproduce

Use config:
```yaml
liip_imagine:
    driver: "vips"
```

### Expected result
Can use `vips` driver.

### Actual result
App fails with message "Invalid configuration for path "liip_imagine.driver": Invalid imagine driver specified: "vips"".

### PR DESCRIPTION
**Pull request by PabloKowalczyk**
| Q | A
| --- | ---
| Branch? | 2.x
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fix #1494
| License | MIT
| Doc |

### CODE REVIEW

File: DependencyInjection/LiipImagineExtension.php [hunk 1]
```diff
@@ -101,6 +102,17 @@ public function load(array $configs, ContainerBuilder $container)
             $loader->load('templating.xml');
         }
 
+        $driver = $config['driver'];
+        if ($driver === 'vips') {
```
Comment (dbu) [inline DependencyInjection/LiipImagineExtension.php]:
```suggestion
        if ('vips' === $config['driver']) {
```

we could inline the variable here

File: Resources/config/imagine_vips.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<container xmlns="http://symfony.com/schema/dic/services"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+
+        <service id="liip_imagine.vips" class="Imagine\Vips\Imagine" public="false">
```
Comment (dbu) [inline Resources/config/imagine_vips.xml:8]: are these always autoregistered or should we add tags to the service so it always works?

Comment (dbu) [reply Resources/config/imagine_vips.xml:8]: ah sorry, found it. the name is used in https://github.com/liip/LiipImagineBundle/blob/ee1581327e4de483e3359f6bb6dcfe7415bf3042/DependencyInjection/LiipImagineExtension.php#L104 so this is all we need.

**Comment by dbu** [comment]
for phpstan we should add imagine-vips to the require-dev section in composer. (we could also ignore the error, but i prefer adding it, so we notice if something goes off.)

File: DependencyInjection/LiipImagineExtension.php (hunk 1 above)
Comment (PabloKowalczyk) [reply DependencyInjection/LiipImagineExtension.php]: I use it in three other places instead.

**Comment by PabloKowalczyk** [comment]
I've added `rokka/imagine-vips` to `require-dev` and to `suggest` also.

File: DependencyInjection/LiipImagineExtension.php (hunk 1 above)
Comment (dbu) [reply DependencyInjection/LiipImagineExtension.php]: ah, i see. yeah lets keep it then.

File: DependencyInjection/LiipImagineExtension.php [hunk 3]
```diff
@@ -101,13 +102,24 @@ public function load(array $configs, ContainerBuilder $container)
             $loader->load('templating.xml');
         }
 
-        $container->setParameter('liip_imagine.driver_service', 'liip_imagine.'.$config['driver']);
+        $driver = $config['driver'];
+        if ($driver === 'vips') {
```
Comment (dbu) [inline DependencyInjection/LiipImagineExtension.php]:
```suggestion
        if ('vips' === $driver) {
```

File: DependencyInjection/LiipImagineExtension.php [hunk 4]
```diff
@@ -101,13 +102,24 @@ public function load(array $configs, ContainerBuilder $container)
             $loader->load('templating.xml');
         }
 
-        $container->setParameter('liip_imagine.driver_service', 'liip_imagine.'.$config['driver']);
+        $driver = $config['driver'];
+        if ('vips' === $driver) {
+            if (\class_exists(Imagine::class) === false) {
```
Comment (dbu) [inline DependencyInjection/LiipImagineExtension.php]:
```suggestion
            if (!\class_exists(Imagine::class)) {
```

**Comment by dbu** [comment]
ah dang. we support older php than vips does. i will try something with the ci.

**Comment by PabloKowalczyk** [comment]
> ah dang. we support older php than vips does. i will try something with the ci.

PHP 7.1 is quite old https://www.php.net/supported-versions.php :)

**Pull request closed without merging**